### PR TITLE
Metadata improvments

### PR DESCRIPTION
**Pull request by TheMulti0**
I changed a bit of the still 'amazing' upgrade of yesterday.
    • Spitted the `Exceptions.cs` file - into two files, `GameMetadataException.cs`, and `GameNotFoundException.cs`
    • Extracted the 'API' endpoint to a const variable 
    • Improved the `VoteConverter.ReadValue` method readability.

### CODE REVIEW

File: GameMetadata.Net/MetadataClient.cs [hunk 1]
```diff
@@ -6,26 +6,28 @@
...
 
-        protected static T DeserializeObject<T>(string json) =>
-            JsonConvert.DeserializeObject<T>(json, DefaultSettings);
+        private static T DeserializeObject<T>(string json) 
+            => JsonConvert.DeserializeObject<T>(json, DefaultSettings);
+
         public async Task<SearchedGame> GetGameMetadataAsync(string query)
         {
-            var images = await GetAllGameImagesAsync(query);
+            IEnumerable<GameImage> images = await GetAllGameImagesAsync(query);
```
Comment (DynamicField) [inline GameMetadata.Net/MetadataClient.cs:30]: lmao stop ruining my `var`s XD

**Review by DynamicField — APPROVED** [review]
yes

File: GameMetadata.Net/MetadataClient.cs (hunk 1 above)
Comment (TheMulti0) [reply GameMetadata.Net/MetadataClient.cs:30]: ahahah lol

Comment (TheMulti0) [reply GameMetadata.Net/MetadataClient.cs:30]: a. it's just that.
b. why not? - it's not that evident that's it returns an ienumerable

Comment (DynamicField) [reply GameMetadata.Net/MetadataClient.cs:30]: well `AllGamesImages` since it's plural you expect it to be enumerable lol

**Pull request merged**